lgc/disassembler/Disassembler.cpp

```diff
@@ -64,6 +65,62 @@ const char *getPalMetadataRegName(unsigned regNumber);
 
 namespace {
 
+struct InstOp {
+  Optional<int64_t> imm;
+  Optional<unsigned> sReg;
+};
+
+struct InstOrDirective {
+  MCDisassembler::DecodeStatus status = MCDisassembler::Fail;
+  uint64_t offset;
+  ArrayRef<uint8_t> bytes;
+  MCInst mcInst;
+
+  StringRef mnemonic;
+  StringRef comment;
+  InstOp op0;
+  InstOp op1;
+  InstOp op2;
+
+  const MCExpr *valueDirectiveExpr = nullptr;
+
+  uint64_t getEndOffset() const { return offset + bytes.size(); }
+};
+
+class SymbolPool {
+public:
+  SymbolPool(StringSaver &strings, MCContext &context) : m_strings(strings), m_context(context) {}
```

Comment:
It would be nice to have unit tests for this class and be able to quickly see how it works.

Reply:
OK, I will degenerate the class into a function for now then.

Reply:
Done.